primitives/core/src/sr25519.rs

```diff
@@ -537,59 +537,82 @@ impl CryptoType for Pair {
 pub mod vrf {
 	use super::*;
 	#[cfg(feature = "full_crypto")]
-	use crate::crypto::VrfSigner;
-	use crate::crypto::{VrfCrypto, VrfVerifier};
+	use crate::crypto::VrfSecret;
+	use crate::crypto::{VrfCrypto, VrfPublic};
 	use schnorrkel::{
 		errors::MultiSignatureStage,
 		vrf::{VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH},
 		SignatureError,
 	};
 
-	/// VRF transcript ready to be used for VRF sign/verify operations.
-	pub struct VrfTranscript(pub merlin::Transcript);
+	/// VRF input ready to be used for VRF sign and verify operations.
+	// Note: here we may potentially add a closure to check-before-sign...
+	pub struct VrfInput {
+		/// VRF input data.
+		pub data: merlin::Transcript,
+		/// Extra data to be signed by the VRF.
+		pub extra: Option<merlin::Transcript>,
+		/// Optional pre-computed pre-output
+		pub preout: Option<VrfPreOutput>,
+	}
```

Comment:
sr25519 does use merlin transcripts for both VRF inputs and the fiat-shamir transform.  Other VRFs probably won't do this, due to irtf standards, but sr25519 does, and doing so completely is fine for a ristretto based protocol.

Reply:
This type is indeed only for sr25519. Every VRF can define its own with different capabilities, features, etc.

Let's chat about the other point tomorrow